crates/core/types/block.rs

```diff
@@ -211,7 +211,7 @@ pub fn compute_transactions_root(transactions: &[Transaction]) -> H256 {
         // Key: RLP(tx_index)
         // Value: tx_type || RLP(tx)  if tx_type != 0
         //                   RLP(tx)  else
-        trie.insert(idx.encode_to_vec(), tx.encode_to_vec());
+        trie.insert(idx.encode_to_vec(), tx.encode_canonical_to_vec());
```

Comment:
the previous change added a regression to the transaction root calculation. Maybe we can add some unit test for this method to avoid this from happening again in the future?

Reply:
Are we not validating this in EF tests?

Reply:
Good idea

Reply:
The ef tests provide the blocks with the transaction root already set so it is always valid